app/assets/javascripts/admin/addon/components/watched-word-form.js

```diff
@@ -103,29 +105,39 @@ export default class WatchedWordForm extends Component {
         .save()
         .then((result) => {
           this.setProperties({
-            word: "",
+            words: [],
             replacement: "",
             formSubmitted: false,
             selectedTags: [],
             showMessage: true,
             message: I18n.t("admin.watched_words.form.success"),
             isCaseSensitive: false,
           });
-          this.action(WatchedWord.create(result));
-          this.focusInput();
+          if (result.words) {
+            result.words.forEach((word) => {
+              this.action(WatchedWord.create(word));
+            });
+          } else {
+            this.action(result);
+          }
+          // this.focusInput();
         })
         .catch((e) => {
           this.set("formSubmitted", false);
-          const message = e.jqXHR.responseJSON?.errors
-            ? I18n.t("generic_error_with_reason", {
-                error: e.jqXHR.responseJSON.errors.join(". "),
-              })
-            : I18n.t("generic_error");
-          this.dialog.alert({
-            message,
-            didConfirm: () => this.focusInput(),
-            didCancel: () => this.focusInput(),
-          });
+          if (e.jqXHR) {
```

Comment:
All this message stuff should be unnecessary...you can just do `popupAjaxError(e)`.